### PR TITLE
[Xedra Evolved] do smiths dream?

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
Mods "Dreamsmiths got their spells"

#### Purpose of change
So dreamsmiths has almost no spells, and that's bad
#### Describe the solution
The basic idea is this:
- The player has 3-4 tiers of recipes/spells to forge:
- the very basic - stuff you don't even need to forge, you just cover the item (like common knife or pipe) with pure dreamdross (unforged) to obtain some better_than_average melee, but not best one, and you don't need a oneiric hammer for this - just pure mana
- middle tier is truly forged lottery items, most of which uses artifact roll with enchantment properties, so you just forge them to find the one that fit your playstyle the best - warhammer, that boost str, or knife with additional fire damage, or whatever
- and the last tier is hard to obtain, conditional powered thing with unique abilities
- the idea is allow them all to spawn in the world, but second tier is already rare, and the last tier should be extra rare, one per world or similar rarity
Also, to make player not miss his hammer every time, i made the Oneiric Hammer permanent, but it disable your mana usage when you wear it (also we need a flag, that delete the item when it is dropped)

All of it is gain by casting the new Inquire Dreamdross spell - sorta of research system: with some amount of cast you grant to obtain tier 2 recipes, and after long researches you can obtain lvl 3 recipes aslo (lvl 1 recipes should be given from the start)
#### Exact stuff that was added
Tier 1:
knife, club and bo, covered with dreamdross to improve their properties, but without magic

Tier 2:
Dreamforged version of almost every melee from the game (so i think every MA or style got their item to use, from blades to unarmed) and some armor, with random magical properties, powered in the same way as our relics, but without negative effects
Dreamforged ballistic plate, big and medium, both without magical properties

Tier 3:
Helmet, that regen your mana faster the bigger your level of investigation is
Berserking set, powered by berserk knuckles, that boost you the longer you fight with them
Sword, that boost your speed the more enemies is around you
Spear, that can steal some HP from monsters you attack
Quarterstaff, that has constant damage boost, related to how much monsters you slay
"Gunslinger", pair of armguards, that boost your aim

#### TODO

- [X] Make class user autolearn the 1 tier lvl spells
- [X] My greatest enemy - descriptions
- [ ] Maybe some more tier 3 items, but i am out of ideas
- [X] ~~Resolve the bug, where inheritance also copies EoC output, making the EoC hit twice~~ fixed in #63916
- [ ] Fix the fact you can stack the infinite amount of weapon and armor in your inventory, and all of them will boost you

### CODE REVIEW

**Comment by Maleclypse** [comment]
I had been wondering if you could drop procgen enchantments on regular equipment and it's pretty exciting you can!

File: data/mods/Xedra_Evolved/items/melee.json [hunk 1]
```diff
@@ -104,5 +104,309 @@
...
+    "description": "Long quarterstaff, forged entirely from dreamdross.  Twice as heavy than wooden one, but lot sturdier, and still useful as a weapon. The dreamdross grant it some minor magical properties, but you can't control their effects.",
+    "material": [ "forged_dreamstuff" ],
+    "//": "All dreamforged weapons are forged 100% from dreamdross. Wooden parts double their weight, but steel parts almost quarter their weight. 100% steel weapon like swords and knifes has too big benefit from it, so their damage is reduced respectively - the random properties will make them really good weapon anyway, but the basic stats will be sane enough.",
+    "proportional": { "bashing": 1.3, "weight": 2 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_longsword",
+    "copy-from": "longsword",
+    "name": { "str": "dreamforged longsword", "str_pl": "dreamforged longswords" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/melee.json:181]:
```suggestion
    "name": { "str": "dreamforged longsword" },
```

File: data/mods/Xedra_Evolved/items/melee.json [hunk 2]
```diff
@@ -104,5 +104,309 @@
...
+    "description": "Longsword, forged entirely from dreamdross.  It is much lighter than steel one, but it apply much less force to target.  The dreamdross grant it some minor magical properties, but you can't control their effects.",
+    "material": [ "forged_dreamstuff" ],
+    "proportional": { "cutting": 0.5, "weight": 0.25 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_warhammer",
+    "copy-from": "warhammer",
+    "name": { "str": "dreamforged warhammer", "str_pl": "dreamforged warhammers" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/melee.json:190]:
```suggestion
    "name": { "str": "dreamforged warhammer" },
```

File: data/mods/Xedra_Evolved/items/melee.json [hunk 3]
```diff
@@ -104,5 +104,309 @@
...
+    "description": "Warhammer, forged entirely from dreamdross.  The dreamdross grant it some minor magical properties, but you can't control their effects.",
+    "weight": "1367 g",
+    "material": [ "forged_dreamstuff" ],
+    "proportional": { "bashing": 1.1 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_spear",
+    "copy-from": "spear_steel",
+    "name": { "str": "dreamforged spear", "str_pl": "dreamforged spears" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/melee.json:200]:
```suggestion
    "name": { "str": "dreamforged spear" },
```

File: data/mods/Xedra_Evolved/items/melee.json [hunk 4]
```diff
@@ -104,5 +104,309 @@
...
+    "description": "Spear, forged entirely from dreamdross.  The dreamdross grant it some minor magical properties, but you can't control their effects.",
+    "weight": "2129 g",
+    "material": [ "forged_dreamstuff" ],
+    "proportional": { "cutting": 1.1 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_halberd",
+    "copy-from": "halberd",
+    "name": { "str": "dreamforged halberd", "str_pl": "dreamforged halberds" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/melee.json:210]:
```suggestion
    "name": { "str": "dreamforged halberd" },
```

File: data/mods/Xedra_Evolved/items/melee.json [hunk 5]
```diff
@@ -104,5 +104,309 @@
...
+    "description": "Twohand halberd, forged entirely from dreamdross.  The dreamdross grant it some minor magical properties, but you can't control their effects.",
+    "weight": "2374 g",
+    "material": [ "forged_dreamstuff" ],
+    "proportional": { "cutting": 1.1 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_glaive",
+    "copy-from": "glaive",
+    "name": { "str": "dreamforged glaive", "str_pl": "dreamforged glaives" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/melee.json:220]:
```suggestion
    "name": { "str": "dreamforged glaive" },
```

File: data/mods/Xedra_Evolved/items/melee.json [hunk 6]
```diff
@@ -104,5 +104,309 @@
...
+    "description": "Sturdy naginata, forged entirely from dreamdross.  The dreamdross grant it some minor magical properties, but you can't control their effects.",
+    "weight": "2181 g",
+    "material": [ "forged_dreamstuff" ],
+    "proportional": { "cutting": 1.1 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_mace",
+    "copy-from": "mace",
+    "name": { "str": "dreamforged mace", "str_pl": "dreamforged maces" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/melee.json:240]:
```suggestion
    "name": { "str": "dreamforged mace" },
```

File: data/mods/Xedra_Evolved/items/melee.json [hunk 7]
```diff
@@ -104,5 +104,309 @@
...
+    "description": "Light mace, forged entirely from dreamdross.  The dreamdross grant it some minor magical properties, but you can't control their effects.",
+    "weight": "1160 g",
+    "material": [ "forged_dreamstuff" ],
+    "proportional": { "bashing": 0.8 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_morningstar",
+    "copy-from": "morningstar",
+    "name": { "str": "dreamforged morningstar", "str_pl": "dreamforged morningstars" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/melee.json:250]:
```suggestion
    "name": { "str": "dreamforged morningstar" },
```

File: data/mods/Xedra_Evolved/items/melee.json [hunk 8]
```diff
@@ -104,5 +104,309 @@
...
+    "description": "Light morningstar, forged entirely from dreamdross.  So light, that you can't properly apply any force on it.  The dreamdross grant it some minor magical properties, but you can't control their effects.",
+    "weight": "1198 g",
+    "material": [ "forged_dreamstuff" ],
+    "proportional": { "bashing": 0.8 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_estoc",
+    "copy-from": "estoc",
+    "name": { "str": "dreamforged estoc", "str_pl": "dreamforged estocs" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/melee.json:260]:
```suggestion
    "name": { "str": "dreamforged estoc" },
```

File: data/mods/Xedra_Evolved/items/melee.json [hunk 9]
```diff
@@ -104,5 +104,309 @@
...
+    "name": { "str": "dreamforged arming sword", "str_pl": "dreamforged arming swords" },
+    "description": "Arming sword, forged entirely from dreamdross.  So light, that you can't properly apply any force on it.  The dreamdross grant it some minor magical properties, but you can't control their effects.",
+    "material": [ "forged_dreamstuff" ],
+    "proportional": { "cutting": 0.5, "weight": 0.25 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_broadsword",
+    "copy-from": "broadsword",
+    "name": { "str": "dreamforged broadsword", "str_pl": "dreamforged broadswords" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/melee.json:278]:
```suggestion
    "name": { "str": "dreamforged broadsword" },
```

File: data/mods/Xedra_Evolved/items/melee.json [hunk 10]
```diff
@@ -104,5 +104,309 @@
...
+    "name": { "str": "dreamforged estoc" },
+    "description": "Estoc, forged entirely from dreamdross.  So light, that you can't properly apply any force on it.  The dreamdross grant it some minor magical properties, but you can't control their effects.",
+    "material": [ "forged_dreamstuff" ],
+    "proportional": { "cutting": 0.5, "weight": 0.25 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_arming_sword",
+    "copy-from": "arming_sword",
+    "name": { "str": "dreamforged arming sword", "str_pl": "dreamforged arming swords" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/melee.json:269]:
```suggestion
    "name": { "str": "dreamforged arming sword" },
```

File: data/mods/Xedra_Evolved/items/melee.json [hunk 11]
```diff
@@ -104,5 +104,309 @@
...
+    "name": { "str": "dreamforged broadsword", "str_pl": "dreamforged broadswords" },
+    "description": "Arming sword, forged entirely from dreamdross.  So light, that you can't properly apply any force on it.  The dreamdross grant it some minor magical properties, but you can't control their effects.",
+    "material": [ "forged_dreamstuff" ],
+    "proportional": { "cutting": 0.5, "weight": 0.25 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_battleaxe",
+    "copy-from": "battleaxe",
+    "name": { "str": "dreamforged battle axe", "str_pl": "dreamforged battle axes" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/melee.json:287]:
```suggestion
    "name": { "str": "dreamforged battle axe" },
```

File: data/mods/Xedra_Evolved/items/melee.json [hunk 12]
```diff
@@ -104,5 +104,309 @@
...
+    "name": { "str": "dreamforged kris", "str_pl": "dreamforged krises" },
+    "description": "Small curved knife, forged entirely from dreamdross.  You barely feel any weight.  The dreamdross grant it some minor magical properties, but you can't control their effects.",
+    "material": [ "forged_dreamstuff" ],
+    "proportional": { "cutting": 0.3, "weight": 0.25 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_kukri",
+    "copy-from": "kukri",
+    "name": { "str": "dreamforged kukri", "str_pl": "dreamforged kukris" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/melee.json:324]:
```suggestion
    "name": { "str": "dreamforged kukri" },
```

File: data/mods/Xedra_Evolved/items/melee.json [hunk 13]
```diff
@@ -104,5 +104,309 @@
...
+    "name": { "str_sp": "nodachi" },
+    "description": "Long nodachi, forged entirely from dreamdross.  The dreamdross grant it some minor magical properties, but you can't control their effects.",
+    "material": [ "forged_dreamstuff" ],
+    "proportional": { "cutting": 0.3, "weight": 0.25 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_rapier",
+    "copy-from": "rapier",
+    "name": { "str": "dreamforged rapier", "str_pl": "dreamforged rapiers" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/melee.json:342]:
```suggestion
    "name": { "str": "dreamforged rapier" },
```

File: data/mods/Xedra_Evolved/items/melee.json [hunk 14]
```diff
@@ -104,5 +104,309 @@
...
+    "name": { "str_sp": "wakizashi" },
+    "description": "Long nodachi, forged entirely from dreamdross.  The dreamdross grant it some minor magical properties, but you can't control their effects.",
+    "material": [ "forged_dreamstuff" ],
+    "proportional": { "cutting": 0.3, "weight": 0.25 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_zweihander",
+    "copy-from": "zweihander",
+    "name": { "str": "dreamforged zweihänder", "str_pl": "dreamforged zweihänders" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/melee.json:369]:
```suggestion
    "name": { "str": "dreamforged zweihänder" },
```

File: data/mods/Xedra_Evolved/items/melee.json [hunk 15]
```diff
@@ -104,5 +104,309 @@
...
+    "name": { "str": "dreamforged khopesh", "str_pl": "dreamforged khopeshes" },
+    "description": "Light khopesh, forged entirely from dreamdross.  The dreamdross grant it some minor magical properties, but you can't control their effects.",
+    "material": [ "forged_dreamstuff" ],
+    "proportional": { "cutting": 0.3, "weight": 0.25 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_lucerne",
+    "copy-from": "lucern_hammer",
+    "name": { "str": "dreamforged lucerne hammer", "str_pl": "dreamforged lucerne hammers" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/melee.json:387]:
```suggestion
    "name": { "str": "dreamforged lucerne hammer" },
```

**Comment by GuardianDll** [comment]
image

File: data/mods/Xedra_Evolved/items/melee.json [hunk 16]
```diff
@@ -104,5 +104,352 @@
...
+    "copy-from": "dreamforged_broadsword",
+    "name": { "str": "tsurugi of speed", "str_pl": "tsurugi of speed" },
+    "description": "Classic japanese tsurugi sword with razor sharp blade.  You start to think you need to move faster, and effect intensifies when you are surrounded.",
+    "relic_data": { "passive_effects": [ { "id": "MON_NEARBY_SPEED" } ] }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_q_staff_exp",
+    "copy-from": "dreamforged_q_staff",
+    "name": { "str": "bo of practice", "str_pl": "bo of practices" },
```
Comment (GuardianDll) [inline data/mods/Xedra_Evolved/items/melee.json:451]:
```suggestion
    "name": { "str": "bo of practice", "str_pl": "bos of practices" },
```

File: data/mods/Xedra_Evolved/items/armor/armor.json [hunk 17]
```diff
@@ -446,5 +446,283 @@
...
+    "delete": { "flags": [ "ABLATIVE_LARGE" ] },
+    "extend": { "flags": [ "ABLATIVE_MEDIUM" ] }
+  },
+  {
+    "id": "dreamforged_fullhelmet",
+    "copy-from": "tac_fullhelmet",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": { "str": "dreamforged full helmet" },
+    "description": "Manually forged helmet, that resemble the shape of modern army helmets.  This one protect both face and eyes, using a special forged mask, but has no visor, and not so comfortable. The dreamdross grant it some small magical properties, but you can't control their effects.",
```
Comment (GuardianDll) [inline data/mods/Xedra_Evolved/items/armor/armor.json:498]:
```suggestion
    "description": "Manually forged helmet, that resemble the shape of modern army helmets.  This one protect both face and eyes, using a special forged mask, but has no visor, and not so comfortable.  The dreamdross grant it some small magical properties, but you can't control their effects.",
```

File: data/mods/Xedra_Evolved/items/melee.json [hunk 18]
```diff
@@ -104,5 +104,352 @@
...
+    "id": "dreamforged_glaive_healing",
+    "name": { "str": "qiang of life", "str_pl": "spears of life" },
+    "description": "Long, dense spear.  It can steal a life from creature it attacks, healing you for a bit.",
+    "relic_data": { "passive_effects": [ { "id": "ENCHANTMENT_VAMPIRISM" } ] }
+  },
+  {
+    "type": "GENERIC",
+    "id": "dreamforged_tsurugi_speed",
+    "copy-from": "dreamforged_broadsword",
+    "name": { "str": "tsurugi of speed", "str_pl": "tsurugi of speed" },
```
Comment (GuardianDll) [inline data/mods/Xedra_Evolved/items/melee.json:443]:
```suggestion
    "name": { "str_sp": "tsurugi of speed", "str_pl": "tsurugis of speed" },
```

File: data/mods/Xedra_Evolved/spells/dreamsmithing.json [hunk 19]
```diff
@@ -3,24 +3,20 @@
     "id": "spell_oneiric_hammer",
     "type": "SPELL",
     "name": "Call Oneiric Hammer",
-    "description": "Creates a crystal blacksmith's hammer with which to build a new reality.",
+    "description": "Creates a crystal blacksmith's hammer with which to build a new reality.  You need all your magic force to hold it, and you unable to do any magic while using it.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/spells/dreamsmithing.json:6]: Ah thank you, this explains the purpose of the negative -1 max mana above!

**Review by Maleclypse — APPROVED** [review]
You have really outdone yourself with this!  This turned out way better than I think I would have made this. Minor wording changes here and there and most of my questions answered themselves later in the review.

**Pull request merged**